### ISSUE TITLE
Offline/unreachable peer vs. disabled channel?

### ISSUE THREAD

**Issue by xanoni** [edouardparis/lntop#39]
QQ, if `lntop` shows `inactive` as the status of a channel, my understanding is that this could mean one of 4 things:

1a) The other node is offline or some connection problem between them and us (e.g., broken Tor circuit)
1b) We are offline (and maybe them too)
2a) They disabled the channel (`lncli updatechanstatus --action disable xyz xyz`)
2b) We disabled the channel (and maybe them as well)

Would it be possible to differentiate at least some of these cases, given some can be fixed while others can not be fixed by us?

E.g., when I `nmap` [1] the nodes behind channels that are shown as inactive, I noticed that scenario (A) usually means that there's a connection problem, whereas (B) means that one of the parties has disabled the channel (=not much we can do, if it was them).

**(A) Peer is probably offline**
<img width=200 src=https://user-images.githubusercontent.com/77220130/130872640-722c7451-b6e9-412f-9134-8443004e1f2d.png />

**(B) Peer has probably disabled the channel**
<img width=200 src=https://user-images.githubusercontent.com/77220130/130872653-4778d3e1-0cbb-4e58-89ed-2cdcad831076.png />

I'm not sure how much information `lnd` exposes, but if we could at least get the IPs / Onions of our peers, then we could make `lntop` aware of the true connection state and decide what status to show.

[1] Command: `proxychains4 nmap -Pn -p 9735 XYZ.onion`

**Comment by xanoni** [comment]
TBH, I'm no longer too sure if that heuristic above made sense ... might have been coincidence. However, the overall question is still valid :)

**Comment by edouardparis** [comment]
You are right, `active` / `inactive` is not clear about what the actual status of the channel.
`lntop` displays information directly from the rpc call response of lnd:
https://api.lightning.community/#lnrpc-channel
Maybe we can cross multiple rpc response to get more information as you suggest, so we keep the issue open :+1:

**Comment by rkfg** [comment]
`Inactive` only means the peer isn't connected at all. Channel status is a bit tricky because it's not intuitive. A channel can be disabled on any side or on both and if it's disabled on your peer's side it means you can _send_ through this channel (or forward a payment to it) but not _receive_ from it. So it becomes a one-way street. If it's disabled on your side then your peer can send you coins (or forward through you) but you can't use this channel yourself unless you enable it.

There's currently [an issue](https://github.com/lightningnetwork/lnd/issues/6000) in lnd that can cause channels to be disabled even after a short connection interruption and the channel can stay in this state for a while (up to 24 hours) because these updates are rate limited by every node. So if you see that someone disabled an active channel to you it doesn't mean they're mad at you or something :smile: Most probably it's enabled from their point of view but this update hasn't made it to your node yet. Yes, even if you're directly connected it might take hours to propagate, at least that's my experience. I hope when it's fixed these spontaneous disables will go away and the overall update rate will go down and cause less throttling. Btw, if you can find a similar situation in your logs as described in https://github.com/lightningnetwork/lnd/issues/6000, feel free to upvote the report and add your logs so we'll know it's not just me.

Either way, if you see the channel as active the peer is online. If at the same time it's disabled on your side for a while maybe it's time to check the logs for any suspicious messages. If it's disabled on your peer's side there's nothing you can do, just wait it out. Most often it will get re-enabled in a day or less, if it stays disabled and is drained (so it physically can't be refilled and as such is useless) for a week and you can't contact your peer through other means (mail/Twitter/etc.) maybe it's time to close and reopen the channel.
